File: AgentenSimulation_1/src/Mediator.java
import java.io.*;
import java.util.Collections;
import java.util.List;
import java.util.Random;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Mediator {

	int contractSize;
	int generationSize;
	
	public Mediator(int contractSizeA, int contractSizeB, int generationSize) throws FileNotFoundException{
		if(contractSizeA != contractSizeB){
			throw new FileNotFoundException("Verhandlung kann nicht durchgefuehrt werden, da Problemdaten nicht kompatibel");
		}
		this.contractSize = contractSizeA;
		this.generationSize = generationSize;
	}
	
	public Contract[] initContract(){
		Contract[] contracts = new Contract[generationSize];

		List<Integer> contract = IntStream.range(0,contractSize).boxed().collect(Collectors.toList());
		for (int i = 0; i <generationSize; i++) {
			Collections.shuffle(contract);
			contracts[i] = new Contract(contract.stream().mapToInt(Integer::intValue).toArray());
		}

		return contracts;
	}

	public Contract[] getRandomContracts(int count){
		Contract[] contracts = new Contract[count];
		List<Integer> contract = IntStream.range(0,contractSize).boxed().collect(Collectors.toList());
		for (int i = 0; i < count; i++) {
			Collections.shuffle(contract);
			contracts[i] = new Contract(contract.stream().mapToInt(Integer::intValue).toArray());
		}
		return contracts;
	}

	public int[] constructProposal(int[] contract) {
		int[] proposal = new int[contractSize];
		for(int i=0;i<proposal.length;i++)proposal[i] = contract[i];
		int element = (int)((proposal.length-1)*Math.random());
		int wert1   = proposal[element];
		int wert2   = proposal[element+1];
		proposal[element]   = wert2;
		proposal[element+1] = wert1;
		return proposal;
	}
}

File: AgentenSimulation_Rust/src/mediator.rs
use std::io;
use rand::seq::SliceRandom;
use rand::thread_rng;
use crate::contract::Contract;

pub(crate) struct Mediator {
    contract_size: usize,
}

impl Mediator {
    pub fn new(contract_size_a: usize, contract_size_b: usize) -> Result<Self, io::Error> {
        if contract_size_a != contract_size_b {
            return Err(io::Error::new(
                io::ErrorKind::Other,
                "Verhandlung kann nicht durchgefuehrt werden, da Problemdaten nicht kompatibel",
            ));
        }
        Ok(Self {
            contract_size: contract_size_a,
        })
    }

    pub fn get_random_contracts(&self, count: usize) -> Vec<Contract> {
        let mut contracts = Vec::with_capacity(count);
        let mut rng = thread_rng();

        for _ in 0..count {
            let mut contract: Vec<usize> = (0..self.contract_size).collect();
            contract.shuffle(&mut rng);
            contracts.push(Contract {contract});
        }

        contracts
    }
}